firefox_ui_harness/runtests.py

```diff
@@ -29,9 +48,46 @@ def __init__(self, *args, **kwargs):
         self.test_handlers = [FirefoxTestCase]
 
 
-def run():
-    cli(runner_class=ReleaseTestRunner, parser_class=ReleaseTestParser)
+def startTestRunner(runner_class, options, tests):
+    try:
+        # If the specified binary is an installer it needs to be installed
+        installed = False
+        if mozinstall.is_installer(options.binary):
```

Comment:
I guess for this to make sense we'll have to add a condition to make sure they aren't both specified, like you say. So I'll leave it up to you whether this is worth doing right now.